modules/cassandra/src/test/java/org/testcontainers/containers/CassandraContainerTest.java

```diff
@@ -0,0 +1,132 @@
+            assertEquals("Cassandra configuration is not overridden", TEST_CLUSTER_NAME_IN_CONF, resultSet.one().getString(0));
+        }
+    }
+
+    @Test(expected = ContainerLaunchException.class)
+    public void testEmptyConfigurationOverride() {
+        try (CassandraContainer cassandraContainer = new CassandraContainer<>()
+                .withConfigurationOverride("cassandra-empty-configuration")
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+        }
+    }
+
+    @Test
+    public void testInitScript() {
+        try (CassandraContainer cassandraContainer = new CassandraContainer<>()
+                .withInitScript("initial.cql")
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            testInitScript(cassandraContainer);
+        }
+    }
+
+    @Test
+    public void testInitScriptWithLegacyCassandra() {
+        try (CassandraContainer cassandraContainer = new CassandraContainer<>("cassandra:2.2.11")
+                .withInitScript("initial.cql")
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            testInitScript(cassandraContainer);
+        }
+    }
+
+    @Test
+    public void testCassandraQueryWaitStrategy() {
+        try (CassandraContainer cassandraContainer = new CassandraContainer<>()
+                .waitingFor(new CassandraQueryWaitStrategy())
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+        }
+    }
+
+    @Test
+    public void testCassandraGetCluster() {
+        try (CassandraContainer cassandraContainer = new CassandraContainer<>()
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            ResultSet resultSet = performQuery(cassandraContainer.getCluster(), "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertNotNull("Result set has no release_version", resultSet.one().getString(0));
+        }
+    }
+
+    private void testInitScript(CassandraContainer cassandraContainer) {
+        ResultSet resultSet = performQuery(cassandraContainer, "SELECT * FROM keySpaceTest.catalog_category");
+        assertTrue("Query was not applied", resultSet.wasApplied());
+        Row row = resultSet.one();
+        assertEquals("Inserted row is not in expected state", 1, row.getLong(0));
+        assertEquals("Inserted row is not in expected state", "test_category", row.getString(1));
+    }
+
+    private ResultSet performQuery(CassandraContainer cassandraContainer, String cql) {
+        Cluster explicitCluster = Cluster.builder()
```

Comment:
`cassandraContainer.getCluster()` ?

Reply:
No, it's intentional. As you can see there are two performQuery methods. First builds cluster instance with general information from container (container ip and mapped port) and invokes second performQuery method which just uses passed cluster argument to perform query. 
Second method is used directly in testCassandraGetCluster() test which creates cluster instance via getCluster() method and invokes performQuery() method.

Two ways of connection configuration are tested here.